drivers/soundwire/intel_auxdevice.c

```diff
@@ -82,7 +82,9 @@ static void generic_new_peripheral_assigned(struct sdw_bus *bus,
 	}
 
 	if (sdw->link_res->hw_ops->program_sdi)
-		sdw->link_res->hw_ops->program_sdi(sdw, dev_num);
+		if (bus->dev_num_alloc == SDW_DEV_NUM_ALLOC_IDA_WAKE_ONLY &&
```

Comment:
Do we need to call program_sdi if dev_num_alloc = SDW_DEV_NUM_ALLOC_IDA, too?

Reply:
we could, yes.

Reply:
done